From e53d46d90f0b73be34c42abfd759b16306da03ac Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Jul 2026 11:08:42 +0200
Subject: [PATCH] evaluate: fix check for unknown in cmd_op_to_name
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

JIRA: https://issues.redhat.com/browse/RHEL-190549
Upstream Status: nftables commit 9f76bb63c0c706ea5c0d55931ee690ca5dccaf16

commit 9f76bb63c0c706ea5c0d55931ee690ca5dccaf16
Author: 谢致邦 (XIE Zhibang) <Yeking@Red54.com>
Date:   Wed Feb 7 15:10:20 2024 +0000

    evaluate: fix check for unknown in cmd_op_to_name

    Example:
    nft --debug=all destroy table ip missingtable

    Before:
    Evaluate unknown

    After:
    Evaluate destroy

    Fixes: e1dfd5cc4c46 ("src: add support to command "destroy"")
    Signed-off-by: 谢致邦 (XIE Zhibang) <Yeking@Red54.com>
    Signed-off-by: Phil Sutter <phil@nwl.cc>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 src/evaluate.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/evaluate.c b/src/evaluate.c
index 98f97bd..20950fd 100644
--- a/src/evaluate.c
+++ b/src/evaluate.c
@@ -5886,7 +5886,7 @@ static const char * const cmd_op_name[] = {
 
 static const char *cmd_op_to_name(enum cmd_ops op)
 {
-	if (op > CMD_DESCRIBE)
+	if (op >= array_size(cmd_op_name))
 		return "unknown";
 
 	return cmd_op_name[op];
